source/common/config/subscription_factory_impl.cc

```diff
@@ -77,6 +77,14 @@ SubscriptionPtr SubscriptionFactoryImpl::subscriptionFromConfigSource(
       CustomConfigValidatorsPtr custom_config_validators =
           std::make_unique<CustomConfigValidatorsImpl>(validation_visitor_, server_,
                                                        api_config_source.config_validators());
+
+      JitteredExponentialBackOffStrategyPtr backoff_strategy;
+      auto& grpc_services = api_config_source.grpc_services();
+      if (!grpc_services.empty() && grpc_services[0].has_envoy_grpc()) {
+        backoff_strategy = Utility::prepareJitteredExponentialBackOffStrategy(
+            grpc_services[0].envoy_grpc(), api_.randomGenerator(), absl::nullopt, absl::nullopt);
+      }
```

Comment:
This code seems to be repeating itself a few times here.
I think it can be moved to an unnamed namespace function in this file, and also ensure that a backoff-strategy is always created (either with the given config, or default values). This should simplify the `GrpcStream` c'tor.

Reply:
EDIT: looks like this function repeats itself in another file (see below),so I think it should be static-public.

Reply:
moved the repeated code to source/common/config/utility.h